src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -630,9 +630,11 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
 
         val owner = lifetime.ancestorStrict<RsGenericDeclaration>() ?: return
         val declarationParts = listOfNotNull(
-            owner.typeParameterList,
+            (owner as? RsImplItem)?.traitRef,
+            (owner as? RsImplItem)?.typeReference,
+            (owner as? RsFunction)?.typeParameterList,
             (owner as? RsFunction)?.valueParameterList,
-            owner.whereClause
+            (owner as? RsFunction)?.whereClause
```

Comment:
I suppose we should check `whereClause` of `impl` blocks as well since:
```rust
#![feature(in_band_lifetimes)]
trait T {}
struct S<'a>(&'a str);
impl T for S<'_> where 'a : 'static {}
```
is allowed and 
```rust
#![feature(in_band_lifetimes)]
trait T {}
struct S<'a>(&'a str);
impl T for S where 'a : 'static {}
```
produces only 
```
error[E0726]: implicit elided lifetime not allowed here
 --> src/main.rs:4:12
  |
4 | impl T for S where 'a : 'static {}
  |            ^- help: indicate the anonymous lifetime: `<'_>`
```

Reply:
Fixed.